src/toil/job.py

```diff
@@ -1384,12 +1385,16 @@ def __init__(self, service):
         self.jobWrapper = None
 
     def run(self, fileStore):
+        
+        # we need access to the filestore from underneath the service job
+        self.fileStore = fileStore
```

Comment:
The fact that you have to do this is a bad smell, coming from Toil. JobFunctionWrappingJob does the same assignment but other Job subclasses don't, most likely because they don't need it. We should decide if we pass the fileStore down or if we set it on the job *before* `run()` is being invoked.